### DISCUSSION TITLE
PaddleOCR/latest/ppocr/quick_start

### DISCUSSION THREAD

**Discussion by giscus** [PaddlePaddle/PaddleOCR#14137] [Q&A]
# PaddleOCR/latest/ppocr/quick_start

Awesome multilingual OCR toolkits based on PaddlePaddle (practical ultra lightweight OCR system, support 80+ languages recognition, provide data annotation and synthesis tools, support training and deployment among server, mobile, embedded and IoT devices)

https://paddlepaddle.github.io/PaddleOCR/latest/ppocr/quick_start.html

**Comment by 202230483077yyh** [comment]
这里cuda支持12版本吗？还是一定得cuda11？

**Reply by GreatV** [reply]
可以，具体参看paddlepaddle官网的安装指南

**Comment by zwzypmmlc** [comment]
请问识别结果能区分不同颜色字体吗？

**Reply by GreatV** [reply]
不能

**Comment by MrJson1** [comment]
使用滑动窗口进行检测和识别 运行直接报错了，最新版本paddleocr-2.9.1，报错新增如下：
Traceback (most recent call last):
  File "E:\pythonProjects\PaddleOCR-main\版本测试\test02.py", line 23, in <module>
    results = ocr.ocr(img_path, cls=True, slice=slice)
TypeError: ocr() got an unexpected keyword argument 'slice'

**Reply by GreatV** [reply]
无法复现

```python
from paddleocr import PaddleOCR
from PIL import Image, ImageDraw, ImageFont
# 初始化OCR引擎
ocr = PaddleOCR(use_angle_cls=True, lang="en")
img_path = "./333783499-13eb31f6-f92f-4902-865a-7722c6a6409c.jpg"
slice = {'horizontal_stride': 300, 'vertical_stride': 500, 'merge_x_thres': 50, 'merge_y_thres': 35}
results = ocr.ocr(img_path, cls=True, slice=slice)
# 加载图像
image = Image.open(img_path).convert("RGB")
draw = ImageDraw.Draw(image)
font = ImageFont.truetype("./doc/fonts/simfang.ttf", size=20)  # 根据需要调整大小
# 处理并绘制结果
for res in results:
    for line in res:
        box = [tuple(point) for point in line[0]]  # 将列表转换为元组列表
        box = [(min(point[0] for point in box), min(point[1] for point in box)),
               (max(point[0] for point in box), max(point[1] for point in box))]
        txt = line[1][0]
        draw.rectangle(box, outline="red", width=2)  # 绘制矩形
        draw.text((box[0][0], box[0][1] - 25), txt, fill="blue", font=font)  # 在矩形上方绘制文本
# 保存结果
image.save("result.jpg")
```

![result](https://github.com/user-attachments/assets/95cc3765-2ae0-40a8-b3dd-93f020f2a062)

**Comment by yuluzhong** [comment]
npu不支持ppocr直接命令行使用吗

**Reply by SWHL** [reply]
刚看了一下，的确不支持

**Comment by 9963KK** [comment]
命令行的输出结果如何可视化为文件呢？

**Reply by SWHL** [reply]
哪个命令行的输出结果呀？

**Comment by Dawei-Simayi** [comment]
解析pdf的结果怎么转成json啊，或者直接转成markdown

**Comment by yy7054wyq5** [comment]
支持商用吗？

**Reply by SWHL** [reply]
支持的，项目是Apache 2.0的，商业友好型开源协议

**Comment by hcysct** [comment]
同时识别中英文

**Reply by hcysct** [reply]
如何设置同时识别中英文

**Reply by SWHL** [reply]
默认ch模型就同时支持中英文的

**Comment by Hulenkius** [comment]
图片中有成行的字，我想要输出每一个单字的文本框数据，这可以吗？

**Reply by MY-SY** [reply]
我也想问，请问你找到解决方法了吗

**Comment by xyqiu902022** [comment]
能否指定按方向识别，比如按列识别？

**Reply by SWHL** [reply]
这个应该是没有这个参数的。是否按列识别主要是看文本检测能否检测到竖着的文本框

**Reply by xyqiu902022** [reply]
好的

**Comment by Bruce-Zhr** [comment]
怎么把中间层的结果hook出来啊，我想看看经过Global Mix Blocks后的结果，一直hook不出来啊
